File: option.py
import sys
import os
import math

class Calculator:
    @staticmethod
    def get_input(prompt=""):
        """Get user input with a prompt."""
        return input(prompt)

    @staticmethod
    def ask_to_continue():
        """Ask the user if they want to quit, continue or return to the menu."""
        print("Do you want to quit or go to the menu? (Enter 'y' to quit, 'n' to continue, 'm' for menu):")
        user_input = Calculator.get_input()
        if user_input == "y":
            print("Goodbye :)")
            sys.exit(0)
        elif user_input == "n":
            return True
        elif user_input == "m":
            option.show_menu()
            return False
        else:
            print("Invalid input. Please try again.")
            return False

class Sum(Calculator):
    @staticmethod
    def perform():
        """Perform the addition operation."""
        while True:
            print("Enter the numbers to add: ")
            a = float(Calculator.get_input())
            print("+")
            b = float(Calculator.get_input())
            result = a + b
            print(f"= {result}")
            if not Calculator.ask_to_continue():
                return

class Difference(Calculator):
    @staticmethod
    def perform():
        """Perform the subtraction operation."""
        while True:
            print("Enter the numbers to subtract: ")
            a = float(Calculator.get_input())
            print("-")
            b = float(Calculator.get_input())
            result = a - b
            print(f"= {result}")
            if not Calculator.ask_to_continue():
                return

class Division(Calculator):
    @staticmethod
    def perform():
        """Perform the division operation."""
        while True:
            print("Enter the numbers to divide: ")
            a = float(Calculator.get_input())
            print("/")
            b = float(Calculator.get_input())
            if b == 0:
                print("Cannot divide by zero. Please try again.")
                continue
            result = a / b
            print(f"= {result}")
            if not Calculator.ask_to_continue():
                return

class Multiplication(Calculator):
    @staticmethod
    def perform():
        """Perform the multiplication operation."""
        while True:
            print("Enter the numbers to multiply: ")
            a = float(Calculator.get_input())
            print("x")
            b = float(Calculator.get_input())
            result = a * b
            print(f"= {result}")
            if not Calculator.ask_to_continue():
                return

class EvenOdd(Calculator):
    @staticmethod
    def perform():
        """Check if a number is even or odd."""
        while True:
            print("Enter the number to check:")
            number = int(Calculator.get_input())
            result = "even" if number % 2 == 0 else "odd"
            print(f"{number} is {result}.")
            if not Calculator.ask_to_continue():
                return

class Factorial(Calculator):
    @staticmethod
    def perform():
        """Calculate the factorial of a number."""
        while True:
            print("Enter the number to calculate its factorial:")
            number = int(Calculator.get_input())
            result = math.factorial(number)
            print(f"Factorial of {number} = {result}")
            if not Calculator.ask_to_continue():
                return

class Option:
    @staticmethod
    def show_menu():
        """Display the main menu and handle user choices."""
        print("Let's do some calculations!")
        print("What do you want to do?")
        print("Here are your options:")
        print("0 - Exit")
        print("1 - Sum")
        print("2 - Difference")
        print("3 - Divide")
        print("4 - Multiply")
        print("5 - Check even or odd")
        print("6 - Calculate factorial")
        print("Enter any number listed above to continue:")

        choice = Calculator.get_input()

        if choice == "0":
            print("Goodbye! :)")
            sys.exit(0)
        elif choice == "1":
            Sum.perform()
        elif choice == "2":
            Difference.perform()
        elif choice == "3":
            Division.perform()
        elif choice == "4":
            Multiplication.perform()
        elif choice == "5":
            EvenOdd.perform()
        elif choice == "6":
            Factorial.perform()
        else:
            print("Invalid input. Please try again.")
            Option.show_menu()

# Start the program
Option.show_menu()

File: option.kt
import java.math.BigInteger
import java.util.*

internal object index {
    val sc = Scanner(System.`in`)
}

fun main() {
    option.main()
}

internal object option {
    @JvmStatic
    fun main(args: Array<String>) {
        showMainMenu()
    }

    // Function to display the main menu and handle user input
    private fun showMainMenu() {
        println("Let's do some calculations!")
        println("What do you want to do?")
        println("Here are your options:")
        println("0 - Exit")
        println("1 - Sum")
        println("2 - Difference")
        println("3 - Divide")
        println("4 - Multiply")
        println("5 - Check even or odd")
        println("6 - Calculate factorial")
        println("Enter a number listed above to continue:")

        val choice = sc.next()

        when (choice) {
            "0" -> exitProgram()
            "1" -> performOperation("add")
            "2" -> performOperation("subtract")
            "3" -> performOperation("divide")
            "4" -> performOperation("multiply")
            "5" -> checkEvenOdd()
            "6" -> calculateFactorial()
            else -> {
                println("Invalid input. Please try again.")
                showMainMenu()
            }
        }
    }

    // Exit the program
    private fun exitProgram() {
        println("Goodbye! :)")
        System.exit(0)
    }

    // Ask if the user wants to continue, return to the menu, or exit
    private fun shouldContinue(): Boolean {
        println("Do you want to quit (y), continue (n), or return to the menu (m)?")
        val choice = sc.next().lowercase()
        return when (choice) {
            "y" -> {
                println("Goodbye! :)")
                System.exit(0)
                false
            }
            "n" -> true
            "m" -> {
                showMainMenu()
                false
            }
            else -> {
                println("Invalid input, returning to menu.")
                showMainMenu()
                false
            }
        }
    }

    // Perform addition, subtraction, multiplication, and division
    private fun performOperation(operation: String) {
        val num1 = getUserInput("Enter the first number:")
        val num2 = getUserInput("Enter the second number:")
        val result = when (operation) {
            "add" -> num1 + num2
            "subtract" -> num1 - num2
            "multiply" -> num1 * num2
            "divide" -> {
                if (num2 == 0f) {
                    println("Division by zero is not allowed.")
                    performOperation("divide") // Retry division
                    return
                }
                num1 / num2
            }
            else -> 0f
        }

        println("The result is: $result")
        if (!shouldContinue()) return
    }

    // Check if a number is even or odd
    private fun checkEvenOdd() {
        println("Enter a number to check:")
        val number = sc.nextInt()
        println("$number is ${if (number % 2 == 0) "even" else "odd"}.")
        if (!shouldContinue()) return
    }

    // Calculate factorial
    private fun calculateFactorial() {
        println("Enter a number to calculate its factorial:")
        val number = sc.nextLong()
        var factorial = BigInteger.ONE
        for (i in 1..number) {
            factorial = factorial.multiply(BigInteger.valueOf(i))
        }
        println("The factorial of $number is: $factorial")
        if (!shouldContinue()) return
    }

    // Helper function to get user input for numbers
    private fun getUserInput(prompt: String): Float {
        println(prompt)
        return sc.nextFloat()
    }
}
